xarray/core/utils.py

```diff
@@ -83,9 +81,10 @@ def maybe_coerce_to_str(index, original_coords):
 
     pd.Index uses object-dtype to store str - try to avoid this for coords
     """
+    from .dtypes import result_type as dtypes_result_type
 
     try:
-        result_type = dtypes.result_type(*original_coords)
+        result_type = dtypes_result_type(*original_coords)
```

Comment:
No strong reason really, `result_type` was the only thing used from `dtypes`. But then I saw it would be redefining itself inside the try/except and thought it's safer to just alias it. We can revert to the original if you prefer?


You can see the error in 2c11948 where it errored at `import xarray`. Not completely sure why it worked before, now that pycompat was imported at init this issue came into light? They seem clearly cyclic now when I look at it though:
https://github.com/pydata/xarray/blob/b1bd6c8236cc070d3cdece0b52746825b4c280cd/xarray/core/utils.py#L34
https://github.com/pydata/xarray/blob/b1bd6c8236cc070d3cdece0b52746825b4c280cd/xarray/core/dtypes.py#L5

And this is what `util` had to be imported for:
https://github.com/pydata/xarray/blob/b1bd6c8236cc070d3cdece0b52746825b4c280cd/xarray/core/dtypes.py#L8

Reply:
I think it makes sense to move the import of `dtypes` to the function (to break the cyclic import), but I'd suggest to revert to importing the whole module:
```suggestion
        result_type = dtypes.result_type(*original_coords)
```

Reply:
Done.